### ISSUE TITLE
Huge memory consumption when importing large obj files

### ISSUE THREAD

**Issue by MinecraftPlaye** [godotengine/godot#44242]
**Godot version:**
3.2.3

**OS/device including version:**
OS: ArchLinux
Linux Kernel: 5.4.80-2-lts
GPU: RX Vega 64
drivers: mesa-20.3.0-3
backend: GLES3

**Issue description:**
I imported a ~600MB OBJ file in Godot. It filled up nearly my entire 32GB of RAM. Importing the same 3D model in Blender only consumes 4GB of RAM. Exporting it in Blender to glTF 2.0 and then importing that in Godot does not cause this issue and can currently be used as a workaround.

**Steps to reproduce:**
1. Download the data from this blog: 
https://ingowald.blog/2020/10/10/sharing-my-nasa-mars-lander-unstructured-mesh-data-set/
2. Download `umesh`
3. Run `./umeshExtractIsoSurface /space/lander-small-rho-9000.umesh --obj iso.obj --iso 0.2`
4. Import `iso.obj`

**Comment by Calinou** [comment]
Could you upload the final OBJ file somewhere? Building `umeshExtractIsoSurface` Feel free to create a dummy GitHub repository in which you create a release (which lets you upload any binary file up to 2 GB).

**Comment by fire** [comment]
Which one is /space/lander-small-rho-9000.umesh ?

**Comment by MinecraftPlaye** [comment]
> Could you upload the final OBJ file somewhere, please? Building `umeshExtractIsoSurface` requires the CUDA toolkit which makes it a nontrivial task.
> 
> Feel free to create a dummy GitHub repository in which you create a release (which lets you upload any binary file up to 2 GB).

I tried to create a GitHub repository and upload the file, but it did not let me upload any files larger than 100MB. Thought I [uploaded the obj file to Google drive](https://drive.google.com/file/d/1vJEXW8rjGiu9R1Ip8_r-w6CljhXcSpBk/view?usp=sharing). 


> Which one is /space/lander-small-rho-9000.umesh ?
> 
> The small dataset is 9GB.

Yes, it is the small dataset. Compressed 9GB, uncompressed 14GB and exported to OBJ 600MB.

**Comment by fire** [comment]
I can confirm that after loading the model, I use 38 gigabytes of ram as the steady state, not sure about Godot Engine exactly though.

Although the spaceship uses plenty of memory, the obj doesn't crash Godot Engine which is good. The result appears in Godot Engine.

**Comment by fire** [comment]
Just for fun, I took the obj and imported the spacecraft into Blender. I then exported it as glb.

Afterwards, I used mesh optimizer.

`./gltfpack -i spacecraft.glb -o spacecraft-opt.glb -si 0.01 -sa` from 1.5gb glb to 81mb` 2.9mb.

[lander-opt.zip](https://github.com/godotengine/godot/files/5673913/lander-opt.zip)

We are planning to use https://twitter.com/reduzio/status/1336307093473734662 for optimizations.

**Comment by MinecraftPlaye** [comment]
> Although the spaceship uses plenty of memory, the obj doesn't crash Godot Engine which is good. The result appears in Godot Engine.
Yep. But with 32GB of RAM for the entire system, it really becomes sluggish as stuff is moved out into Swap.
Thought I'm curious what Blender does differently from Godot as Blender does not even nearly take up that much RAM when importing the obj as Godot does. I assume they are doing some optimizations to the data when importing it as the RAM usage grows (never hitting 16GB, thought) and after completing the import of the obj the RAM usage drops down to less than 5GB. 

> Just for fun, I took the obj and imported the spacecraft into Blender. I then exported it as glb.
> 
> Afterwards, I used mesh optimizer.
> 
> `./gltfpack -i spacecraft.glb -o spacecraft-opt.glb -si 0.01 -sa` from 1.5gb glb to 2.9mb.
> 
> [lander-opt.zip](https://github.com/godotengine/godot/files/5673913/lander-opt.zip)
> 
> We are planning to use https://twitter.com/reduzio/status/1336307093473734662 for optimizations.
> 
> ![editor_screenshot_2020-12-10T074836-0800](https://user-images.githubusercontent.com/32321/101796001-5a1cd100-3abd-11eb-9bba-cfd4e46cf813.png)

Interesting. I will definitely take a look at that tool. :) Thought the settings used for the obj in `lander-opt.zip` might have been a bit too aggressive as a lot of the details are gone when you look at the back of the model.

**Comment by MinecraftPlaye** [comment]
> I can confirm that after loading the model, I use 38 gigabytes of ram as the steady state, not sure about Godot Engine exactly though.
> 
> Although the spaceship uses plenty of memory, the obj doesn't crash Godot Engine which is good. The result appears in Godot Engine.

I tested it again and I can confirm that the engine itself takes way less RAM than the editor. I also tested it with a larger data set (2.4gb glb file) and the editor straight out crashes until the model is removed from the file path.
